### PR TITLE
Update Australia holidays: add `en_AU`, `en_US`, `th` localization

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Adds **`en_AU`**, `en_US`, `th` localization for Australia holidays.

Thai translation has been cross-checked with the available state-level sources whenever applicable [[1]](https://business.vic.gov.au/business-information/public-holidays/public-holidays-in-languages-other-than-english/thai) and a fellow Thai developer ( @ansntu ) who currently resides there.

> [!NOTE]
> Queensland does, in fact, have [different "Royal Queensland Show"/similar events holiday dates depending on the regions/cities you're in](https://www.qld.gov.au/recreation/travel/holidays/show) . For simplicity's sake the current implementation will adhere to [the main Public Holidays list](https://www.qld.gov.au/recreation/travel/holidays/public) for now and keep those aforementioned entries for possible future implementations.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/australia.py [hunk 1]
```diff
@@ -415,21 +440,22 @@ def _populate_subdiv_nt_public_holidays(self):
 
     def _populate_subdiv_nt_half_day_holidays(self):
         if self._year >= 2016:
-            begin_time_label = "%s (from 7pm)"
+            # %s (from 7pm).
+            begin_time_label = tr("%s (from 7pm)")
```
Comment (KJhellico) [inline holidays/countries/australia.py:444]:
```suggestion
            begin_time_label = self.tr("%s (from 7pm)")
```

File: holidays/countries/australia.py [hunk 2]
```diff
@@ -415,21 +440,22 @@ def _populate_subdiv_nt_public_holidays(self):
...
 
             # Christmas Eve.
-            self._add_christmas_eve(begin_time_label % "Christmas Eve")
+            self._add_christmas_eve(begin_time_label % tr("Christmas Eve"))
```
Comment (KJhellico) [inline holidays/countries/australia.py:447]:
```suggestion
            self._add_christmas_eve(begin_time_label % self.tr("Christmas Eve"))
```

File: holidays/countries/australia.py [hunk 3]
```diff
@@ -415,21 +440,22 @@ def _populate_subdiv_nt_public_holidays(self):
...
 
             # New Year's Eve.
-            self._add_new_years_eve(begin_time_label % "New Year's Eve")
+            self._add_new_years_eve(begin_time_label % tr("New Year's Eve"))
```
Comment (KJhellico) [inline holidays/countries/australia.py:450]:
```suggestion
            self._add_new_years_eve(begin_time_label % self.tr("New Year's Eve"))
```

File: holidays/countries/australia.py [hunk 4]
```diff
@@ -661,20 +687,21 @@ def _populate_subdiv_sa_public_holidays(self):
 
     def _populate_subdiv_sa_half_day_holidays(self):
         if self._year >= 2012:
-            begin_time_label = "%s (from 7pm)"
+            # %s (from 7pm).
+            begin_time_label = tr("%s (from 7pm)")
```
Comment (KJhellico) [inline holidays/countries/australia.py:691]:
```suggestion
            begin_time_label = self.tr("%s (from 7pm)")
```

File: holidays/countries/australia.py [hunk 5]
```diff
@@ -661,20 +687,21 @@ def _populate_subdiv_sa_public_holidays(self):
...
 
             # Christmas Eve.
-            self._add_christmas_eve(begin_time_label % "Christmas Eve")
+            self._add_christmas_eve(begin_time_label % tr("Christmas Eve"))
```
Comment (KJhellico) [inline holidays/countries/australia.py:694]:
```suggestion
            self._add_christmas_eve(begin_time_label % self.tr("Christmas Eve"))
```

File: holidays/countries/australia.py [hunk 6]
```diff
@@ -661,20 +687,21 @@ def _populate_subdiv_sa_public_holidays(self):
...
 
             # New Year's Eve.
-            self._add_new_years_eve(begin_time_label % "New Year's Eve")
+            self._add_new_years_eve(begin_time_label % tr("New Year's Eve"))
```
Comment (KJhellico) [inline holidays/countries/australia.py:697]:
```suggestion
            self._add_new_years_eve(begin_time_label % self.tr("New Year's Eve"))
```

**Review by KJhellico — COMMENTED** [review]
Localization of new country is as good as new country! 🚀

**Review by KJhellico — APPROVED** [review]
LGTM.

**Review by arkid15r — COMMENTED** [review]
@PPsyrius great work!
@ansntu thanks for helping with this!

**Pull request merged**